rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,1197 @@
+                        title="hibernate Search javadoc" />
+                    <tag>hibernate-search</tag>
+                    <tag>annotation</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00108">
+            <when>
+                <javaclass references="org.hibernate.search.annotations.CacheFromIndex">
+                    <location>ANNOTATION</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Deprecated annotation CacheFromIndex in Hibernate Search 5" effort="1" severity="optional">
+                    <message>Remove the `@CacheFromIndex` annotation. No alternative replacement necessary.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_changes"
+                        title="Changes in Hibernate Search 5.x" />
+                    <link href="https://docs.jboss.org/hibernate/search/5.5/api/org/hibernate/search/annotations/CacheFromIndex.html"
+                        title="hibernate Search javadoc for @CacheFromIndex" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00109">
+            <when>
+                <javaclass references="org.hibernate.search.annotations.Key">
+                    <location>ANNOTATION</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Deprecated annotation Key in Hibernate Search 5" effort="1" severity="optional">
+                    <message>Custom filter cache keys are a deprecated feature and are scheduled to be removed in Hibernate Search 6.
+                     As of Hibernate Search 5.1, the filter cache keys are determined automatically based on the filter parameters so it is no longer required to provide a key object. </message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <link href="http://hibernate.org/search/documentation/migrate/5.1/#api-changes" title="Hibernate Search 5.1 API changes" />
+                    <link href="https://docs.jboss.org/hibernate/search/5.5/api/index.html?org/hibernate/search/annotations/Key.html"
+                        title="Hibernate Search javadoc for @Key" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00110">
+            <when>
+                <javaclass references="org.hibernate.search.backend.configuration.impl.IndexWriterSetting.{constant}" />
```

Comment:
This doesn't seem to fire for me. The test only uses an `import` of this (which is wrong BTW and couldn't even compile, it only makes sense as `import static`), which isn't very realistic. My test has this:

```java
import org.hibernate.search.backend.configuration.impl.IndexWriterSetting;

...
        System.out.println(IndexWriterSetting.TERM_INDEX_INTERVAL);
...
```

That's not very realistic either :-), but should be a good model of how Java usage of this could look like. It also seems that these `IndexWriterSetting`-s are often used as keys to a map of settings (or possibly even in Hibernate configuration XML/.properties).